ajuna-test/src/sidechain.rs

```diff
@@ -0,0 +1,237 @@
+		Block = MockBlock<SideChainRuntime>,
+		NodeBlock = MockBlock<SideChainRuntime>,
+		UncheckedExtrinsic = MockUncheckedExtrinsic<SideChainRuntime>,
+	{
+		System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+		AjunaBoard: pallet_ajuna_board::{Pallet, Call, Storage, Event<T>},
+	}
+);
+
+parameter_types! {
+	pub const BlockHashCount: u64 = 250;
+	pub const SS58Prefix: u8 = 42;
+}
+
+type BlockNumber = u64;
+
+impl frame_system::Config for SideChainRuntime {
+	type BaseCallFilter = frame_support::traits::Everything;
+	type BlockWeights = ();
+	type BlockLength = ();
+	type DbWeight = ();
+	type Origin = Origin;
+	type Call = Call;
+	type Index = u64;
+	type BlockNumber = BlockNumber;
+	type Hash = H256;
+	type Hashing = BlakeTwo256;
+	type AccountId = AccountId;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type Header = Header;
+	type Event = Event;
+	type BlockHashCount = BlockHashCount;
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = ();
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type SS58Prefix = SS58Prefix;
+	type OnSetCode = ();
+	type MaxConsumers = frame_support::traits::ConstU32<16>;
+}
+
+parameter_types! {
+	pub const MaxNumberOfPlayers: u8 = 2;
+}
+
+pub type Guess = u32;
+
+use crate::traits::BlockProcessing;
+use frame_support::{pallet_prelude::MaxEncodedLen, RuntimeDebugNoBound};
+use scale_info::TypeInfo;
+
+const MAX_PLAYERS: usize = 2;
+
+#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebugNoBound, TypeInfo, MaxEncodedLen)]
+pub struct GameState {
+	pub players: [AccountId; MAX_PLAYERS],
+	pub next_player: u8,
+	pub solution: Guess,
+	pub winner: Option<AccountId>,
+}
+
+// Rules
+// One player can only have one go at a time
+// It's a guessing game where a player has to guess the right number
+// Initial state will have this number
+
+pub const THE_NUMBER: Guess = 42;
+
+pub struct NumberGame;
+
+impl TurnBasedGame for NumberGame {
+	type State = GameState;
+	type Player = AccountId;
+	type Turn = Guess;
+
+	fn init(players: &[Self::Player]) -> Option<Self::State> {
+		if players.len() != MAX_PLAYERS {
+			return None
+		};
+
+		let mut p = vec![[0u8; 32].into(); MAX_PLAYERS];
+		p.clone_from_slice(players);
+		Some(GameState {
+			players: <[AccountId; MAX_PLAYERS]>::try_from(p).unwrap(),
+			next_player: 0,
+			solution: THE_NUMBER,
+			winner: None,
+		})
+	}
+
+	fn play_turn(
+		player: Self::Player,
+		state: Self::State,
+		turn: Self::Turn,
+	) -> Option<Self::State> {
+		if state.winner.is_some() {
+			return None
+		}
+
+		if !state.players.contains(&player) {
+			return None
+		}
+
+		if state.players[state.next_player as usize] != player {
+			return None
+		}
+
+		let mut state = state;
+		state.next_player = (state.next_player + 1) % state.players.len() as u8;
+
+		if state.solution == turn {
+			state.winner = Some(player);
+		}
+
+		Some(state)
+	}
+
+	fn is_finished(state: &Self::State) -> Finished<Self::Player> {
```

Comment:
Opps I missed this before but having a separate `fn finish` might be more explicit:
```rust
fn is_finished(state: &Self::State) -> bool { ... }
fn finished(state: &Self::State) -> Finished<Self::Player> { ... }
```